### ISSUE TITLE
Period Tracker to detect PCOS

### ISSUE THREAD

**Issue by hit-haa18** [SrijanShovit/HealthLearning#51]
Hi @SrijanShovit 
The purpose of this issue is to build and enhance period tracker application to better support individuals with PCOS or to detect PCOS.
Could you assign me this issue under GSSOC'24?

**Comment by SrijanShovit** [comment]
Which dataset are you going to use?
Explain the steps of your approach. I shall assign then to you.

**Comment by hit-haa18** [comment]
Dataset: https://www.kaggle.com/datasets/prasoonkottarathil/polycystic-ovary-syndrome-pcos
or https://www.kaggle.com/datasets/sahilkoli04/pcos2023
There is no particular dataset with all the characteristics but these two are close and could be improved further by user inputs
Approach :
Preprocess the data by handling missing values, encoding categorical variables, and normalizing numerical features. Choose a suitable machine learning algorithm for binary classification, considering models like logistic regression, decision trees, random forests, or gradient boosting algorithms.
Split the dataset into training and validation sets using techniques like train-test split or k-fold cross-validation.Train the model and fine-tune it.
Use evaluation metrics like accuracy, F1-score to assess the model's performance on the validation set and fine-tune hyperparameters as needed. And explore feature engineering.

**Comment by SrijanShovit** [comment]
Good. Just check once the dataset being used in PCOS detection of this repo is different or same. Moreover, would like to know you said "Period Tracker" , so is it some what different from detecting PCOS?

**Comment by hit-haa18** [comment]
> Good. Just check once the dataset being used in PCOS detection of this repo is different or same. Moreover, would like to know you said "Period Tracker" , so is it some what different from detecting PCOS?

The dataset is not same 
Should the same dataset be used ?

And about the period tracker I think integrating PCOS prediction capabilities into a period tracker application could provide users with additional insights into their reproductive health and help identify potential indicators of PCOS and warn them in advance based on their on inputs.

**Comment by SrijanShovit** [comment]
No, it's ok if you want to use other dataset. Only thing is since PCOS detection is already going, Period Tracker should be a bit different, additive. Anyways, decide upon the features on which you want to track periods. That's a very good idea  and appreciate you think for females. Start the ML pipelines only after architecture is prepared.

**Comment by hit-haa18** [comment]
Ok sure.
How would you like me to proceed?

**Comment by SrijanShovit** [comment]
Think about some plan first. Check which datasets or their combination works. What you wanna ask from people on your app as input data. Once you express that I will provide my inputs.

**Comment by KomalSrivastava** [comment]
@SrijanShovit Sir I want to work on this project could you please kindly assign me this project. The goal of this initiative is to develop and improve a period tracker application to provide better support for individuals with PCOS or to aid in the detection of PCOS.

**Comment by SrijanShovit** [comment]
@hit-haa18 Any updates? Otherwise I will let @KomalSrivastava proceed.

**Comment by hit-haa18** [comment]
> @hit-haa18 Any updates? Otherwise I will let @KomalSrivastava proceed.

I want to use the dataset I provided before itself and take inputs from user like duration,bp,Recent Weight Gain,hair growth,skin darkening,hair loss,Acne/ Pimples
could you give me your insights on how to proceed and assign me this issue?

**Comment by SrijanShovit** [comment]
Let's have a meet today evening 7 pm to discuss this. Just DM me once on discord.

**Comment by hit-haa18** [comment]
> Let's have a meet today evening 7 pm to discuss this. Just DM me once on discord.

Can we have it around 5 pm ?

**Comment by SrijanShovit** [comment]
ok sure

**Comment by Taranpreet10451** [comment]
Can you please assign this to me.

**Comment by hit-haa18** [comment]
> ok sure 

When can we have the meet?
